File: content/search/source/includes/view/code-snippets/csharp/DecimalToDoubleCreateIndex.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Connect to your Atlas deployment
var uri = "<connectionString>";
var client = new MongoClient(uri);
var database = client.GetDatabase("sample_airbnb");
var collection = database.GetCollection<BsonDocument>(
    "listings_SearchablePrice");

// Define your MongoDB Search index
var index = new CreateSearchIndexModel(
  "listingsSearchablePrice", new BsonDocument
  {
    { "mappings", new BsonDocument
      {
        { "dynamic", true }
      }
    }
  });

// Create the index
var result = collection.SearchIndexes.CreateOne(index);
Console.WriteLine($"New index name: {result}");

File: content/search/source/includes/view/code-snippets/java/DecimalToDoubleCreateIndex.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class DecimalToDoubleCreateIndex {
    public static void main(String[] args) {
        // Connect to your Atlas deployment
        String uri = "<connectionString>";
        try (MongoClient mongoClient = MongoClients.create(uri)) {
            MongoDatabase database = mongoClient.getDatabase("sample_airbnb");
            MongoCollection<Document> collection =
                database.getCollection("listings_SearchablePrice");

            // Define your MongoDB Search index
            Document definition = new Document("mappings",
                new Document("dynamic", true));

            // Create the index
            String result = collection.createSearchIndex(
                "listingsSearchablePrice", definition);
            System.out.println("New index name: " + result);
        }
    }
}
